library/cmac.c

```diff
@@ -339,7 +340,7 @@ int mbedtls_cipher_cmac_finish( mbedtls_cipher_context_t *ctx,
     last_block = cmac_ctx->unprocessed_block;
 
     /* Calculate last block */
-    if( cmac_ctx->padding_flag )
+    if( cmac_ctx->padding_flag && cmac_ctx->unprocessed_len < block_size )
```

Comment:
If `cmac_ctx->unprocessed_len == block_size` then you don't need to pad. That's why we have the additional condition.